plugins/worker/lib/timeout.js

```diff
@@ -92,28 +95,62 @@ async function process(timeoutConfig, buildId, redis) {
  * If yes, abort build.
  * @method check
  * @param {Object} redis
+ * @param {Object} redlock
  * @return {Promise}
  */
-async function check(redis) {
-    const keys = await redis.hkeys(`${queuePrefix}timeoutConfigs`);
+async function check(redis, redlock, workerId) {
+    const keys = await redis.hkeys(hash);
 
     if (!keys || keys.length === 0) return;
 
     await Promise.all(
         keys.map(async buildId => {
-            const json = await redis.hget(`${queuePrefix}timeoutConfigs`, buildId);
+            try {
+                const lock = await redlock.lock(buildId, REDIS_LOCK_TTL);
```

Comment:
instead of locking each build, shall we just lock on `hash` which is `${queuePrefix}timeoutConfigs` so one worker processes all builds exclusively ?

Reply:
We can't lock a Redis hash, can only lock a key.
```Failed to lock timeoutConfigs Exceeded 10 attempts to lock the resource "timeoutConfigs".```

Reply:
just have a dummy key for sake of locking ?

Reply:
you will have to create it first